includes/adapter/RecipientFromWCOrder.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\WooCommerce\includes\adapter;
+
+use MyParcelNL\Sdk\src\Helper\ValidateStreet;
+use MyParcelNL\Sdk\src\Model\Recipient;
+use WCMYPA_Admin;
+use WCMYPA_Settings;
+use WPO\WC\MyParcel\Compatibility\Order as WCX_Order;
+
+abstract class RecipientFromWCOrder extends Recipient
+{
+    public const SUFFIX_CHECK_REG = "~^([a-z]{1}\d{1,3}|-\d{1,4}\d{2}\w{1,2}|[a-z]{1}[a-z\s]{0,3})(?:\W|$)~i";
+
+    /**
+     * RecipientFromWCOrder constructor.
+     *
+     * @throws \Exception
+     */
+    public function __construct()
+    {
+        $recipientDetails = $this->prepareOrderData();
+        parent::__construct($recipientDetails, $this->carrierCountry);
+    }
+
+    abstract public function prepareOrderData(): array;
+
+    abstract public function getAddressParts();
+
+    /**
+     * @param  $order
+     *
+     * @return string|null
+     */
+    public function getEmailAddress($order): ?string
+    {
+        $deliveryOptions = WCX_Order::get_meta($order, WCMYPA_Admin::META_DELIVERY_OPTIONS);
```

Comment:
Liever het "echte" delivery options ding gebruiken, niet handmatig in die array gaan lopen graven. `WCMYPA_Admin::getDeliveryOptionsFromOrder($order)`

Reply:
^